projects/common-consumption/src/lib/content-details/toc-item/toc-item.component.ts

```diff
@@ -34,19 +37,18 @@ export class TocItemComponent implements OnInit, OnChanges {
   @Output() noContent: EventEmitter<any> = new EventEmitter();
   @Output() selectedItem: EventEmitter<any> = new EventEmitter();
   @Output() playButtonClick: EventEmitter<any> = new EventEmitter();
-
+  private resetActiveContent = false;
   get FlattenedType() { return FlattenedType; }
   get MimeTypeMasterData() { return MimeTypeMasterData; }
-
+  get ExpandBehavior() { return ExpandBehavior; }
   isMimeTypeFilterChanged = false;
-
   private isSameMimeTypeInChildren = ((mimeTypesCount, activeMimeType) => {
     const contentMimeType = Object.keys(JSON.parse(mimeTypesCount));
     return Boolean(activeMimeType.filter(value => contentMimeType.includes(value)).length);
   });
   private isSameMimeType = (currentContent) => this.activeMimeTypeFilter.find(mimeType => mimeType === currentContent.mimeType);
   private isChildrenPresent = (currentContent) => Boolean(currentContent.children && currentContent.children.length);
-
+  private subscriptions: Subscription[] = [];
```

Comment:
ngOnDestroy() {
    this.subscriptions.forEach(s => s.unsubscribe());
  }
Rather than have mulitple '.unsubscribe()' statements, I used one statement for 'unsubscribe'. This is used for only one subscription at the moment for filter changes. Any future subscriptions need to just append to this array instead of multiple private properties to maintain subscriptions and multiple '.unsubscribe()' statments on ngOnDestroy()